### PR TITLE
Allow overriding MaxItemCount for query/changefeed in Spark connector

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**

### CODE REVIEW

**Review by kushagraThapar — APPROVED** [review]
public API changes look good to me.

File: sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/configuration-reference.md [hunk 1]
```diff
@@ -19,6 +19,7 @@ Configuration Reference:
 | :---        |    :----   |         :--- | 
 | `spark.cosmos.useGatewayMode`      | `false`    | Use gateway mode for the client operations  |
 | `spark.cosmos.read.forceEventualConsistency`  | `true`    | Makes the client use Eventual consistency for read operations instead of using the default account level consistency |
+| `spark.cosmos.read.maxItemCount`  | `1000`    | Overrides the maximum number of documents that can be returned for a single query- or change feed request. The default value is `1000` - consider increasing this only for average document sizes significantly smaller than 1KB.  |
```
Comment (xinlian12) [inline sdk/cosmos/azure-cosmos-spark_3-1_2-12/docs/configuration-reference.md:22]: why 1000? I thought SDK by page default is 100

**Review by xinlian12 — APPROVED** [review]
LGTM. Thanks

**Pull request merged**